app/models/state_file_base_intake.rb

```diff
@@ -407,12 +412,24 @@ def latest_submission
   end
 
   def increment_failed_attempts
+    # If unlock_in time has passed, reset the failed_attempts. Otherwise, the intakes' failed_attempts will
+    # remain unchanged until the verification code is answered correctly (see: reset_failed_attempts_for_login!)
+    # taken from https://github.com/heartcombo/devise/issues/4679#issuecomment-1453450974
+    if last_failed_attempt_at && last_failed_attempt_at < self.class.unlock_in.ago
```

Comment:
looking at the _previous_ failed attempt and seeing if that was more than 30 minutes ago. If so, we can reset the attempts before we increment & check if we need to lock again here

Reply:
can we just use `locked_at`?